Assets/LeapMotionModules/Interaction/InteractionBehaviour.cs

```diff
@@ -556,8 +593,130 @@ public enum GraspedMovementType {
 
     private InteractionHand _closestHoveringHand = null;
 
-    public float GetDistance(Vector3 worldPosition) {
-      return Vector3.Distance(this.transform.position, worldPosition);
+    public virtual float GetComparativeHoverDistance(Vector3 worldPosition) {
+      float closestComparativeColliderDistance = float.PositiveInfinity;
+      bool hasColliders = false;
+      Collider collider;
+      Transform colliderTransform;
+      foreach (var colliderTransformPair in _colliderTransformPairs) {
+        if (!hasColliders) hasColliders = true;
+
+        collider = colliderTransformPair.collider;
+        colliderTransform = colliderTransformPair.transform;
+
+        float testDistance;
+        if (collider is SphereCollider) {
+          testDistance = GetComparativeCenterSqrDistance((SphereCollider)collider, colliderTransform, worldPosition);
+        }
+        else if (collider is CapsuleCollider) {
+          testDistance = GetComparativeCenterSqrDistance((CapsuleCollider)collider, colliderTransform, worldPosition);
+        }
+        else if (collider is BoxCollider) {
+          testDistance = GetComparativeCenterSqrDistance((BoxCollider)collider, colliderTransform, worldPosition);
+        }
+        else if (collider is MeshCollider) {
+          testDistance = GetComparativeCenterSqrDistance((MeshCollider)collider, colliderTransform, worldPosition);
+        }
+        else {
+          testDistance = (this.transform.position - worldPosition).sqrMagnitude;
+        }
+
+        if (testDistance < closestComparativeColliderDistance) {
+          closestComparativeColliderDistance = testDistance;
+        }
+      }
+
+      if (!hasColliders) {
+        return (this.transform.position - worldPosition).sqrMagnitude;
+      }
+      else {
+        return closestComparativeColliderDistance;
+      }
+    }
+
+    /// <summary>
+    /// Returns the squared distance to the center of the argument SphereCollider.
+    /// </summary>
+    protected static float GetComparativeCenterSqrDistance(SphereCollider sphere, Transform sphereTransform, Vector3 worldPosition) {
+      return (sphereTransform.TransformPoint(sphere.center) - worldPosition).sqrMagnitude;
+    }
+
+    /// <summary>
+    /// Returns the squared distance to the line segment at the center of the CapsuleCollider.
+    /// </summary>
+    protected static float GetComparativeCenterSqrDistance(CapsuleCollider capsule, Transform capsuleTransform, Vector3 worldPosition) {
+      // https://docs.unity3d.com/ScriptReference/CapsuleCollider-direction.html
+      Vector3 dir = capsule.direction == 0 ? Vector3.right : capsule.direction == 1 ? Vector3.up : Vector3.forward;
+
+      Vector3 a = capsule.center + dir * (capsule.height / 2F - capsule.radius);
+      Vector3 b = capsule.center - dir * (capsule.height / 2F - capsule.radius);
+      return (worldPosition.ConstrainToSegment(capsuleTransform.TransformPoint(a), capsuleTransform.TransformPoint(b)) - worldPosition)
```

Comment:
You should only need one transform point, since you can just negate the offset from the center to one end to get the offset to the other end (in world space).